### PR TITLE
Content Lifecycle sources will no longer include cloned channels by other projects

### PR DESCRIPTION
**Pull request by lneves12**
## What does this PR change?

Content Lifecycle sources will no longer include cloned channels by other projects

## GUI diff

No difference.

- [ ] **DONE**

## Documentation
- No documentation needed.

- [ ] **DONE**

## Test coverage
- No tests:

- [ ] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/8092

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"		 
- [ ] Re-run test "java_pgsql_tests"		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"

### CODE REVIEW

**Comment by lneves12** [comment]
ping @mseidl @hustodemon or @ncounter

File: java/spacewalk-java.changes [hunk 1]
```diff
@@ -1,3 +1,4 @@
+- Content Lifecycle sources will no longer include cloned channels by other projects (bsc#1137965)
```
Comment (ncounter) [inline java/spacewalk-java.changes]: As per the message needs to be readable and clear to customer as well, I'd rephrase it a bit in a more generic way:

`Hide channels managed by Content Lifecycle projects from available sources (bsc#1137965)`

**Comment by lneves12** [comment]
After discussion with @hustodemon we decided to change the filter implementation logic to the backend. This needs to be reviewed :)

File: java/code/src/com/suse/manager/webui/controllers/shared/channels/ChannelsUtils.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.suse.manager.webui.controllers.shared.channels;
```
Comment (hustodemon) [inline java/code/src/com/suse/manager/webui/controllers/shared/channels/ChannelsUtils.java]: Hmm, we already have a `com.suse.manager.webui.controllers.utils`, wouldn't it make sense to move this class there?

File: java/spacewalk-java.changes [hunk 3]
```diff
@@ -3,6 +3,7 @@
 - Enable product detection for plain rhel systems (bsc#1136301)
 - For orphan contentsources, look also in susesccrepositoryauth to make sure they are not being referenced (bsc#1138275)
 - Fallback to logged-in-user org and then vendor errata when looking up erratum on cloning (bsc#1137308)
+- Hide channels managed by Content Lifecycle projects from available sources (bsc#1137965)
```
Comment (hustodemon) [inline java/spacewalk-java.changes]: Style nitpick: we usually push the new lines to the top of the file.

File: web/spacewalk-web.changes [hunk 4]
```diff
@@ -1,6 +1,7 @@
 - Add new validation to avoid creating content lifecycle projects starting with a number (bsc#1139493)
 - Redirect to first step of channel assignment after change channel submit (bsc#1137244)
-- add error messages, hints for image profile edit page (bsc#1127319) 
+- Hide channels managed by Content Lifecycle projects from available sources (bsc#1137965)
```
Comment (hustodemon) [inline web/spacewalk-web.changes]: Style nitpick: we usually push the new lines to the top of the file.

File: java/code/src/com/suse/manager/webui/controllers/shared/channels/ChannelsUtils.java (hunk 2 above)
Comment (lneves12) [reply java/code/src/com/suse/manager/webui/controllers/shared/channels/ChannelsUtils.java]: Hum I really like to split things by feature/area rather than by function. For me it makes more sense to have something like:

channels
    - utils.java
    - logic.java
another_feature
    - utils.java
    - logic.java


instead of:

utils
    channels.utils.java
    another_feature.utils.java
    another_feature2.utils.java


what do you think ?

Comment (hustodemon) [reply java/code/src/com/suse/manager/webui/controllers/shared/channels/ChannelsUtils.java]: Yeah, those are 2 views to look at that. Yours is totally valid (although it'd lead to explosion of number of packages - mostly we would have a package for each feature that would only contain single controller...). I'd prefer sticking to the convention as we already have:

`webui.controllers`:
- `ImageController`
- `AKController`
- `webui.controllers.utils`:
  - `ImageUtils`
  - `AKUtils`

I wouldn't like mixing those 2 approaches - we should stick to one. Feel free to propose refactoring the `webui.controllers` package on the mailing list. Anyways - this should be solved separately from this bugfix.

Comment (lneves12) [reply java/code/src/com/suse/manager/webui/controllers/shared/channels/ChannelsUtils.java]: I moved everything to "controllers/channels" does that look better?

It makes sense to have everything together that corresponds to one feature :)

The controllers folder is growing uncontrolled and it's getting hard to understand :D
